clientcredentialsx/auth.go

```diff
@@ -0,0 +1,70 @@
+package clientcredentialsx
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/spf13/pflag"
+	"github.com/spf13/viper"
+
+	"go.infratographer.com/x/viperx"
+
+	"golang.org/x/oauth2/clientcredentials"
+)
+
+// AuthConfig is an oauth2 client config
+type AuthConfig struct {
+	// ClientID is the applications's ID
+	ClientID string
+
+	// ClientSecret is the application's secret
+	ClientSecret string
+
+	// TokenURL is the resource server's token endpoint URL. This is a constant specific to each server.
+	TokenURL string
+
+	// Scopes specifies optional requested permissions
+	Scopes []string
+}
+
+// Auth handles client credentials oauth2 authentication
+type Auth struct {
+	cfg clientcredentials.Config
+}
+
+// NewAuth returns a new oauth2 http client
+func NewAuth(cfg AuthConfig) *Auth {
+	ccCfg := clientcredentials.Config{
+		ClientID:     cfg.ClientID,
+		ClientSecret: cfg.ClientSecret,
+		TokenURL:     cfg.TokenURL,
+		Scopes:       cfg.Scopes,
+	}
+
+	cli := &Auth{
+		cfg: ccCfg,
+	}
+
+	return cli
+}
+
+// HTTPClient returns an http client using the configured client credentials config.
+// The token will auto-refresh as necessary.
+func (a Auth) HTTPClient(ctx context.Context) *http.Client {
+	return a.cfg.Client(ctx)
+}
```

Comment:
I almost wonder if it would be better to either provide an `oauth2.TokenSource` implementation here (even if it's still the same one from the `clientcredentials` package) or somehow leverage [`NewClient`](https://pkg.go.dev/golang.org/x/oauth2#NewClient) here. An "Auth" construct works, more or less, but it feels like it sits between the `oauth2` package's `TokenSource` interface and the `NewClient` logic as opposed to complementing one of the two.